### ISSUE TITLE
How to change the bounding box thickness and label with the demo

### ISSUE THREAD

**Issue by Kingbadger3d** [pjreddie/darknet#1045]
Hey,

Anybody know how to make the bounding box thinner and label smaller in the example app displayed with opencv. Cheers J Also when I run the example app the video runs at the same frame rate that the detector is working (7 fps in my case on 750ti), but ive seen other examples where the video frame rate is correct even when the detection fps is low like 7-8.

Im new so please forgive me asking for help here.

**Comment by PeterQuinn925** [comment]
Yeah, write yourself a program to call Yolo and use the results to draw your own bounding box. Example here: https://github.com/PeterQuinn925/Squirrel/blob/master/sort_photos.py

**Comment by Kingbadger3d** [comment]
Thanks Peter, Hahaha I like the squirrel deterrent app I was thinking of doing something simular with video detection to alert of grey squirels in red squirel zones.

Would it be possible to do this directly in the C / demo framework or do you think python would be better, also how easy would it be to port your script to work on video in/webcam.

Thanks for the help mate

**Comment by PeterQuinn925** [comment]
I don't know how easy it would be in the C/demo framework. I am much better with Python than with C, so I ran with that. 

In terms of getting it to read from a webcam - trivial. You just need to use opencv to read the webcam into an image instead of reading from a file.  Something like (but not exactly): 

cam = cv2.VideoCapture(0)
s,img = cam.read()

There are tons of examples for getting images from webcams using OpenCV2.

**Comment by mathieuorhan** [comment]
hi @Kingbadger3d,

You can modify **image.c** in function **draw_detections_cv_v3** line 464 to fit your need. For instance mine is two times thiner than the original :
`int width = show_img->height * .003;`


In the same function you can modify other display parameters such as font size.

**Comment by nuannuan1991** [comment]
Dear mathieuorhan,
Can you tell me how to modify the program so that the label of the predict bounding box does not display?

**Comment by TolgaBkm** [comment]
> hi @Kingbadger3d,
> 
> You can modify **image.c** in function **draw_detections_cv_v3** line 464 to fit your need. For instance mine is two times thiner than the original :
> `int width = show_img->height * .003;`
> 
> In the same function you can modify other display parameters such as font size.

Thank you for your answer. I had to run `make` again after modifying **image.c**.

**Comment by JavierTSalas** [comment]
If you are wanting to change the font size for `darknet detector XXXXX` you should navigate to the **detector.c** to find the specific implementation for XXXX.
For `darknet detector demo` I navigated to **detector.c** which brought me to **demo.c** which brought me to **image_opencv.cpp** where I could make the following changes

**src/image_opencv.cpp line 944**
`float const font_size = show_img->rows / 1000.F; ` <= This is the default value 
`float const font_size  = 0.45f;` <= You can overwrite it with a specific value.

You can also set the label size in this file.

**NOTE** I would NOT recommend doing this unless you know that all of your classes are the same fixed size. Since we are hardcoding the font_size, all labels will be this size regardless of how big or small the bounding box is.

**Comment by suhaibashraf** [comment]
> Hey,
> 
> Anybody know how to make the bounding box thinner and label smaller in the example app displayed with opencv. Cheers J Also when I run the example app the video runs at the same frame rate that the detector is working (7 fps in my case on 750ti), but ive seen other examples where the video frame rate is correct even when the detection fps is low like 7-8.
> 
> Im new so please forgive me asking for help here.

Hi @Kingbadger3d 
you can find a file **image_opencv.cpp** inside **darknet/src** folder 

To change the bounding box boundaries think or thin, modify the value of **width** at the line **914** 
`int width = std::max(1.0f, show_img->rows * .002f);`

To change the font of the label, modify the value of **font_size** at the line **960**
`float const font_size = show_img->rows / 1000.F;`

To change the size of the bounding box, modify the values of **left, right, top, bot** at lines **944-947**
`int left = (b.x - b.w / 2.)*show_img->cols;`
`int right = (b.x + b.w / 2.)*show_img->cols;`
`int top = (b.y - b.h / 2.)*show_img->rows;`
`int bot = (b.y + b.h / 2.)*show_img->rows;`
